src/EventProducerService.cpp

```diff
@@ -43,11 +43,12 @@ byte EventProducerService::createDefaultEvent(byte evValue)
   unsigned int nodeNum = module_config->nodeNum;
   
   byte index = module_config->findEventSpace();
-  //eventNum = (int)index + 1;
   for (eventNum = 1; eventNum <= module_config->EE_MAX_EVENTS; eventNum++)
   {
     if (module_config->findExistingEvent(nodeNum, eventNum) == module_config->EE_MAX_EVENTS)
     {
+      controller->sendCMDERR(CMDERR_TOO_MANY_EVENTS);
+      controller->sendGRSP(OPC_RQNPN, getServiceID(), CMDERR_TOO_MANY_EVENTS);
```

Comment:
Oops. I actually misread this code. findExistingEvent() returns EE_MAX_EVENTS when it cannot find the event. So in this case this is all OK and this is the ID we should use.

The error condition where the event table is full can be checked after setting index on line 45. I.e. index == EE_MAX_EVENTS.

So please remove the error messages on line 50 and 51. Or add a check for index==EE_MAX_EVENTS and put these error messages there.
I guess that we should increment the error count. But we haven't added these yet for EPService.